rero_ils/modules/acq_accounts/api.py

```diff
@@ -61,49 +64,333 @@ class AcqAccount(IlsRecord):
+        if ref:
+            return extracted_data_from_ref(ref, data='record')
+
+    @property
+    def depth(self):
+        """Get the depth of this account into the account structure.
+
+        As an account can have a parent account, it's interesting to know the
+        depth of this account. A root account has a depth of 0 ; each child has
+        a depth of parent_depth + 1.
+        """
+        parent = self.parent
+        return parent.depth + 1 if parent else 0
+
+    @property
+    def is_active(self):
+        """Get if the account should be considerate as active.
+
+        To know if an account is active, we need to check the related budget.
+        This budget has a 'is_active' field.
+        """
+        from ..budgets.api import BudgetsSearch
+        budget_id = extracted_data_from_ref(self.get('budget'))
+        es = BudgetsSearch() \
+            .filter('term', pid=budget_id) \
+            .source(['is_active']).scan()
+        return next(es).is_active or False
+
+    @property
+    def encumbrance_amount(self):
+        """Get the encumbrance amount related to this account.
+
+        This amount is the sum of :
+          * Order lines related to this account depending of order lines
+            status ; only "pending" order lines must be used to compute this
+            amount.
+          * Encumbrance of all children account
+
+        @:return A tuple of encumbrance amount : First element if encumbrance
+                 for this account, second element is the children encumbrance.
+        """
+        # Encumbrance of this account
+        query = AcqOrderLinesSearch().filter('term', acq_account__pid=self.pid)
+        if AcqOrderLineStatus.OPEN:
+            pending_status = []
+            for status in AcqOrderLineStatus.OPEN:
+                pending_status.append(Q('term', status=status))
+            query = query.query('bool', should=pending_status)
+
+        query.aggs.metric('total_amount', 'sum', field='total_amount')
+        results = query.execute()
+        self_amount = results.aggregations.total_amount.value
+
+        # Encumbrance of children accounts
+        query = AcqAccountsSearch().filter('term', parent__pid=self.pid)
+        query.aggs.metric('total', 'sum', field='encumbrance_amount.total')
+        results = query.execute()
+        children_amount = results.aggregations.total.value
+
+        return self_amount, children_amount
+
+    @property
+    def expenditure_amount(self):
+        """Get the expenditure amount related to this account.
+
+        This amount is the sum of :
+          * Invoice lines related to this account depending of invoice lines
+            status ; only "received" invoice lines must be used to compute this
+            amount.
+          * expenditure of all children account
+
+        @:return A tuple of expenditure amount : First element if expenditure
+                 for this account, second element is the children expenditure.
```

Comment:
*If expenditure* and *is the children expenditure* are not so much explicit. *If there are expenditure*?